tests/unit/test_table_service.py

```diff
@@ -353,3 +353,64 @@ def test_drop_tables(mocker: MockerFixture, mock_table: Table):
     expected_query = "SELECT * FROM information_schema.tables WHERE table_name = ?"
     cursor_mock.execute.assert_any_call(expected_query, [ts.external_table_name])
     cursor_mock.execute.assert_any_call(expected_query, [ts.internal_table_name])
+
+
+def test_drop_outdated_partitions_table_without_partitions(
+    mocker: MockerFixture, mock_table: Table
+):
+    """
+    Test to check if the function 'drop_outdated_partitions' correctly
+    drops the partitions in the fact table that are outdated.
+    """
+
+    connection = MagicMock()
+    cursor_mock = MagicMock()
+    cursor_mock.execute.return_value = 0
+    connection.cursor.return_value = cursor_mock
+
+    ts = TableService(mock_table, connection)
+
+    with pytest.raises(FireboltError):
+        ts.drop_outdated_partitions()
```

Comment:
let's verify that the error is the one we expect, as we have 3 different FireboltErrors raised in that method.
A simple `assert f"{mock_table} does not exist" in e` will do.

Reply:
Done, added match param to the pytest.raises